src/test/java/com/salesforce/dynamodbv2/UpdateTest.java

```diff
@@ -122,56 +156,77 @@ void updateConditionalOnHkSuccess(TestArgument testArgument) {
         });
     }
 
-
     @ParameterizedTest(name = "{arguments}")
     @ArgumentsSource(DefaultArgumentProvider.class)
-    void updateConditionalOnGsiHkSuccess(TestArgument testArgument) {
+    void updateConditionalOnHkSuccessWithLiterals(TestArgument testArgument) {
         testArgument.forEachOrgContext(org -> {
             testArgument.getAmazonDynamoDb().updateItem(new UpdateItemRequest()
-                .withTableName(TABLE3)
+                .withTableName(TABLE1)
                 .withKey(ItemBuilder.builder(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE)
-                    .rangeKey(S, RANGE_KEY_S_VALUE)
                     .build())
-                .withUpdateExpression("set #indexField = :newValue")
-                .withExpressionAttributeNames(ImmutableMap.of("#indexField", INDEX_FIELD))
-                .withExpressionAttributeValues(ImmutableMap.of(":newValue",
-                    createStringAttribute(INDEX_FIELD_VALUE + TABLE3 + org + "Updated"))));
-            assertEquals(ItemBuilder.builder(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE)
-                .someField(S, SOME_FIELD_VALUE + TABLE3 + org)
-                .indexField(S, INDEX_FIELD_VALUE + TABLE3 + org + "Updated")
-                .rangeKey(S, RANGE_KEY_S_VALUE)
-                .build(),
+                .withUpdateExpression("set " + SOME_FIELD + " = :newValue")
+                .withConditionExpression(HASH_KEY_FIELD + " = :currentValue")
+                .addExpressionAttributeValuesEntry(":currentValue",
+                    createAttributeValue(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE))
+                .addExpressionAttributeValuesEntry(":newValue",
+                    createStringAttribute(SOME_FIELD_VALUE + TABLE1 + org + "Updated")));
+            assertEquals(ItemBuilder
+                    .builder(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE)
+                    .someField(S, SOME_FIELD_VALUE + TABLE1 + org + "Updated")
+                    .build(),
                 getItem(testArgument.getAmazonDynamoDb(),
-                    TABLE3,
+                    TABLE1,
                     HASH_KEY_VALUE,
                     testArgument.getHashKeyAttrType(),
-                    Optional.of(RANGE_KEY_S_VALUE)));
+                    Optional.empty()));
         });
     }
 
     @ParameterizedTest(name = "{arguments}")
     @ArgumentsSource(DefaultArgumentProvider.class)
-    void updateConditionalOnHkWithLiteralsSuccess(TestArgument testArgument) {
+    void updateConditionalOnGsiRkSuccess_usePlaceholder(TestArgument testArgument) {
```

Comment:
Expanded existing test case updateConditionalOnGsiRkSuccess into 2 flavors, one using a placeholder (same as before) and one using the literal